src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -372,8 +407,16 @@ private void ResolveAsymmetricAlgorithm(SecurityKey key, string algorithm, bool
             }
             else if (webKey.Kty == JsonWebAlgorithmsKeyTypes.EllipticCurve)
             {
-                CreateECDsaFromJsonWebKey(webKey, willCreateSignatures);
-                return;
+                _ecdsa = CreateECDsaFromJsonWebKey(webKey, willCreateSignatures);
+                _disposeEcdsa = true;
+                if (_ecdsa != null && validateECDSAKeySize(_ecdsa, algorithm))
```

Comment:
It would be better if CreateECDasFromJsonWebKey calls validateECDSAKeySize before 'new ECDsaCng(cngKey)'  otherwise we will leak memory when we throw here.